Crash on sync

```
Caused by: java.lang.ArrayIndexOutOfBoundsException: length=49; index=49
    at java.util.ArrayList.add(ArrayList.java:468)
    at io.horizontalsystems.bankwallet.modules.transactions.TransactionsInteractor.fetchRates(TransactionsInteractor.kt:112)
    at io.horizontalsystems.bankwallet.modules.transactions.TransactionsPresenter.fetchRatesForRecords(TransactionsPresenter.kt:122)
    at io.horizontalsystems.bankwallet.modules.transactions.TransactionsPresenter.didUpdateRecords(TransactionsPresenter.kt:106)
    at io.horizontalsystems.bankwallet.modules.transactions.TransactionsInteractor$onUpdateCoinCodes$$inlined$forEach$lambda$1.accept(TransactionsInteractor.kt:144)
    at io.horizontalsystems.bankwallet.modules.transactions.TransactionsInteractor$onUpdateCoinCodes$$inlined$forEach$lambda$1.accept(TransactionsInteractor.kt:13)
    at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:63)
	... 9 more



Caused by: java.lang.ArrayIndexOutOfBoundsException: length=49; index=49
    at java.util.ArrayList.indexOf(ArrayList.java:331)
    at java.util.ArrayList.contains(ArrayList.java:314)
    at io.horizontalsystems.bankwallet.modules.transactions.TransactionsInteractor.fetchRates(TransactionsInteractor.kt:107)
    at io.horizontalsystems.bankwallet.modules.transactions.TransactionsPresenter.fetchRatesForRecords(TransactionsPresenter.kt:122)
    at io.horizontalsystems.bankwallet.modules.transactions.TransactionsPresenter.didUpdateRecords(TransactionsPresenter.kt:106)
    at io.horizontalsystems.bankwallet.modules.transactions.TransactionsInteractor$onUpdateCoinCodes$$inlined$forEach$lambda$1.accept(TransactionsInteractor.kt:144)
    at io.horizontalsystems.bankwallet.modules.transactions.TransactionsInteractor$onUpdateCoinCodes$$inlined$forEach$lambda$1.accept(TransactionsInteractor.kt:13)
    at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:63)
	... 9 more
```

--- a/app/src/main/java/io/horizontalsystems/bankwallet/modules/transactions/TransactionsInteractor.kt
+++ b/app/src/main/java/io/horizontalsystems/bankwallet/modules/transactions/TransactionsInteractor.kt
@@ -8,6 +8,8 @@ import io.horizontalsystems.bankwallet.entities.TransactionRecord
 import io.reactivex.Single
 import io.reactivex.disposables.CompositeDisposable
 import io.reactivex.schedulers.Schedulers
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.CopyOnWriteArrayList
 import java.util.concurrent.TimeUnit
 
 class TransactionsInteractor(private val adapterManager: IAdapterManager, private val currencyManager: ICurrencyManager, private val rateManager: RateManager) : TransactionsModule.IInteractor {
@@ -17,7 +19,7 @@ class TransactionsInteractor(private val adapterManager: IAdapterManager, privat
     private val ratesDisposables = CompositeDisposable()
     private val lastBlockHeightDisposables = CompositeDisposable()
     private val transactionUpdatesDisposables = CompositeDisposable()
-    private val requestedTimestamps = mutableMapOf<CoinCode, MutableList<Long>>()
+    private val requestedTimestamps = ConcurrentHashMap<CoinCode, MutableList<Long>>()
 
     override fun initialFetch() {
         onUpdateCoinCodes()
@@ -107,7 +109,7 @@ class TransactionsInteractor(private val adapterManager: IAdapterManager, privat
                 if (requestedTimestamps[coinCode]?.contains(timestamp) == true) continue
 
                 if (!requestedTimestamps.containsKey(coinCode)) {
-                    requestedTimestamps[coinCode] = mutableListOf()
+                    requestedTimestamps[coinCode] = CopyOnWriteArrayList()
                 }
                 requestedTimestamps[coinCode]?.add(timestamp)
 

